trino/trino.go

```diff
@@ -1290,89 +1555,50 @@ type queryResponse struct {
-		}
-
-		metadata, err := parseSpoolingMetadata(typedMetadata)
-		if err != nil {
-			return nil, err
-		}
-		switch segment["type"] {
-		case "inline":
-			decodedBytes, err := base64.StdEncoding.DecodeString(segment["data"].(string))
-
-			if err != nil {
-				return nil, fmt.Errorf("error decoding base64 data in inline segment at index %d: %v", segmentIndex, err)
-			}
-
-			decodedData, err := decodeSegment(decodedBytes, sp.encoding, metadata)
-			if err != nil {
-				return nil, fmt.Errorf("failed to decode inline segment at index %d: %v", segmentIndex, err)
-			}
-
-			queryData = append(queryData, decodedData...)
-		case "spooled":
-			uri, ok := segment["uri"].(string)
-			if !ok || uri == "" {
-				return nil, fmt.Errorf("missing or invalid 'uri' field in spooled segment at index %d", segmentIndex)
-			}
-			ackUri, ok := segment["ackUri"].(string)
-			if !ok || ackUri == "" {
-				return nil, fmt.Errorf("missing or invalid 'ackUri' field in spooled segment at index %d", segmentIndex)
-			}
-			headers, ok := segment["headers"].(map[string]interface{})
-			if !ok {
-				return nil, fmt.Errorf("missing or invalid 'headers' field in spooled segment at index %d", segmentIndex)
-			}
+type spooledMetadata struct {
+	SegmentType SegmentType
+	uri         string
+	ackUri      string
+	encoding    string
+	headers     map[string]interface{}
+	metadata    segmentMetadata
+}
 
-			data, err := sp.fetchSegment(uri, ackUri, headers)
-			if err != nil {
-				return nil, fmt.Errorf("failed to fetch segment from uri '%s' at index %d: %v", uri, segmentIndex, err)
-			}
+func parseSpooledMetadata(segment map[string]interface{}, segmentIndex int, segmentMetadata segmentMetadata, encoding string) (spooledMetadata, error) {
+	result := spooledMetadata{
+		metadata:    segmentMetadata,
+		encoding:    encoding,
+		SegmentType: Spooled,
+	}
 
-			decodedData, err := decodeSegment(data, sp.encoding, metadata)
-			if err != nil {
-				return nil, fmt.Errorf("failed to decode spooled segment at index %d: %v", segmentIndex, err)
-			}
+	var ok bool
+	result.uri, ok = segment["uri"].(string)
+	if !ok || result.uri == "" {
+		return spooledMetadata{}, fmt.Errorf("missing or invalid 'uri' field in spooled segment at index %d", segmentIndex)
+	}
 
-			queryData = append(queryData, decodedData...)
-		}
+	result.ackUri, ok = segment["ackUri"].(string)
+	if !ok || result.ackUri == "" {
+		return spooledMetadata{}, fmt.Errorf("missing or invalid 'ackUri' field in spooled segment at index %d", segmentIndex)
+	}
 
+	result.headers, ok = segment["headers"].(map[string]interface{})
+	if !ok {
+		return spooledMetadata{}, fmt.Errorf("missing or invalid 'headers' field in spooled segment at index %d", segmentIndex)
 	}
 
-	return queryData, nil
+	return result, nil
 }
 
-func parseSpoolingMetadata(metadata map[string]interface{}) (spoolingMetadata, error) {
-	result := spoolingMetadata{
+func parseSpoolingMetadata(metadata map[string]interface{}) (segmentMetadata, error) {
```

Comment:
Rename this to `parseSegmentMetadata`, The name `parseSpoolingMetadata` is confusing with `parseSpooledMetadata`.